pkg/api/steve/proxy/proxy.go

```diff
@@ -145,12 +146,15 @@ func (h *Handler) authLocalCluster(router *gmux.Router) func(rw http.ResponseWri
 
 func (h *Handler) matchManagementCRDs() gmux.MatcherFunc {
 	return func(req *http.Request, match *gmux.RouteMatch) bool {
-		splitPath := strings.Split(req.URL.Path, "/")
-		if len(splitPath) < 3 {
+		gvr, err := gvrFromPath(req.URL.Path)
+		if err != nil {
 			return false
 		}
-		apiGroup := splitPath[2]
-		return apiGroup == managementv3.GroupName
+		// Only match v3 management CRDs that are not in the allow-list.
+		return gvr.Group == managementv3.GroupName &&
+			gvr.Version == managementv3.Version &&
```

Comment:
I would not recommend checking the version here. I can forsee a future issue caused by someone not realizing that v4 management resources (hypothetical, don't exist today) are automatically able to be accessed through kubectl.

Reply:
Good point, but should we not check the version at all? Could there be CRs under a special management/v4 group (having nothing to do with our management resources) which we match unnecessarily?

Reply:
> Good point, but should we not check the version at all?

Yes, that would be my opinion.

>Could there be CRs under a special management/v4 group (having nothing to do with our management resources) which we match unnecessarily?

Given the way that k8s does versioning, I don't that we should support/consider the use case of someone else adding a management/v4 api group and expecting our logic to handle that differently. In addition, IIRC the current production implementation does not check for api version, so this new code is adding an exception for things which are currently being blocked off, which I think is not a good idea unless we have a specific reason.